models/model.py

```diff
@@ -1,11 +1,58 @@
 import torch
+import torchvision
+import numpy as np
 import torch.nn as nn
 import torch.nn.functional as F
+from typing import Tuple, List, Optional, Union
 
-from abc import abstractmethod
+from models.model_interface import IModel
 
-class Model(nn.Module):
+class Model(IModel):
```

Comment:
```suggestion
class CNNModel(IModel):
```
wtedy interfejs mógłby się nazywać Model 😃